app/commands/track/determine_concepts_layout.rb

```diff
@@ -0,0 +1,117 @@
+require 'graph_utils'
+
+# ConceptGraph transforms a track configuration to a form compatible for manipulating for
+# displaying the progression of the track
+class Track
+  class DetermineConceptsLayout
+    include Mandate
+
+    def initialize(track_config)
```

Comment:
So, presuming we pass a `track` into this, we have:
```
track.concept_exercises #=> [Track::ConceptExercise]

# Then for each concept exercise
concept_exercise.taught_concepts #=> [Track::Concept]
concept_exercise.prerequisites #=> [Track::Concept]
```

You can get these set up in the tests using something like:
```
track = create track
strings = create :track_concept, slug: "strings", track: track
booleans = create :track_concept, slug: "booleans", track: track

e1 = create :concept_exercise
e1.taught_concepts << strings

e2 = create :concept_exercise
e2.taught_concepts << booleans

e3 = create :concept_exercise
e3.prerequisites << strings
e3.prerequisites << booleans
```

This currently relies on #152 being merged.